### ISSUE TITLE
Audio static after using eqMac for some time

### ISSUE THREAD

**Issue by eric-hu** [bitgapp/eqMac#27]
After a variable amount of time using eqMac, I get static in my audio. I've seen the time vary between 30 minutes and a few hours, and it seems to happen more often when I use Zoom for work calls. I've also experienced it from only using Youtube.

**Workaround**: I'm able to workaround the static by opening eqMac and switching my audio output device and then switching it back. It goes away for some time but still comes back later.

**Comment by advicepyro** [comment]
Experiencing this as well. In my case, I spend all day on my 2017 MacBook Pro Escape for work, and eqMac is fine there. CPU under frequent load but audio does not crackle.

For my 2018 MacBook Pro (with the T2 chip) I get crackling a few minutes in.

**Comment by eric-hu** [comment]
Mine is a 2019 Macbook Pro with a touchbar too. After my initial post I realize that the time intervals must be shorter, because in a 30 minute call I need to reset my audio at least once. 10-20 min is closer to the lower bound I've seen. The static also gets progressively worse if I don't reset my output device. By that 10-20 min time, I've been hearing it for some time, but it was easier to ignore.

**Comment by gagank1** [comment]
I've also been hearing it on my 2018 Macbook Pro. In 5-10 min I went from not hearing it at all to being sure it wasn't just part of a song.

**Comment by ScrollLocker** [comment]
Yeah, it's happening here too. It's closing unexpectedly when I have to get the headphone off... and also, the audio is crackling here after 30 min playing something

**Comment by nodeful** [comment]
Hey guys. Would be very helpful if you listed what Audio devices are you using (headphones, speakers) their base Sample Rate (you can see it in the Audio MIDI Setup.app) and how it's connected to the mac (USB, Audio jack or Bluetooth) Thanks

**Comment by eric-hu** [comment]
I'm using headphones connected via audio jack to my external monitor (BenQ PD2700U)

Audio MIDI Setup says my monitor has an output format of 2 ch 24-bit Integer 48.0 kHz. When I click through every device in Audio MIDI Setup, it shows a 48 kHz output, except for eqMac which shows 44.1 kHz. eqMac and the built in mic and speakers are 32-bit float, as well.

I tried manually changing eqMac's input and output formats to 48 kHZ in Audio MIDI Setup and eqMac crashed and wouldn't start back up.

**Comment by gagank1** [comment]
I'm using Bose QC35 headphones powered off, connected directly to my Mac's 3.5mm audio jack via audio cable. Both the headphones and eqMac have a sample rate of 44.1 khz, and the rest of the sources (mic and built-in speakers) also show the same rate.

**Comment by nodeful** [comment]
I have exactly the same setup as you @gagank1 Never came across any static noise neither with eqMac2 or this new version :/

**Comment by gagank1** [comment]
Hm you're on MacOS 10.15.3 too? I don't really know the details of how sound processing engines work but my first thought was maybe there's some kind of positive feedback loop that's being triggered by certain frequencies/sounds? I was using the rock preset, if that helps at all.

**Comment by ScrollLocker** [comment]
I'm using a headphone... 41.1 KHz 2ch 32-bit float... Nothing is different except the master stream...

**Comment by advicepyro** [comment]
My setup:
- macOS 10.15.4
- using eqMac with my headset connected via the headphone jack
- "External Headphones" was 48000 Hz

Previously eqMac was at 44000 Hz, but since I've changed this to 48000 Hz. I haven't noticed problems in v0.0.0. I will update this comment to see:

- if setting both eqMac and headphones to 48000 Hz produces any crackling on 0.0.0
- if this resolves crackling on 0.0.1

Also curious if anyone else with a non-T1/T2 Mac experiences crackling; so far that appears to be the common denominator.

**Comment by eric-hu** [comment]
@nodeful I tried plugging in directly into my MacBook Pro today using my headphones (Also Bose QC35, first generation, not QC35-II. Also powered off). I still ran into the same issue.

**Comment by ScrollLocker** [comment]
@advicepyro how did you reversed?

**Comment by advicepyro** [comment]
> @advicepyro how did you reversed the version?

Uninstalled then installed the older release from here https://github.com/bitgapp/eqMac/releases/tag/v0.0.0

Still it doesn't look like reverting the version will fix the issue on newer MacBook Pros.

**Comment by nodeful** [comment]
Hey everyone 👋 
I will be open sourcing the new version of eqMac, therefore, I am closing this issue and repository as I will be transfering [nodeful/eqMac2](https://github.com/nodeful/eqMac2) repo to [bitgapp/eqMac](https://github.com/bitgapp/eqMac)
Please bear with me for the next couple of days as I setup that new repo. The eqMac2 code will be living in the [old-eqMac2](https://github.com/bitgapp/eqMac/tree/old-eqmac2) branch if anyone is interested in referencing it for whatever reason.
Please try out the new version and report bugs to the new repo.
Thanks for using eqMac.
